feature/experimental/p4rt/ate_tests/base_p4rt/base_p4rt_test.go

```diff
@@ -280,19 +280,51 @@ func setupP4RTClient(ctx context.Context, args *testArgs) error {
 	return nil
 }
 
-// Function to compare and check if the expected table is present in RPC ReadResponse
-func verifyReadReceiveMatch(expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
+// Function to compare and check if the expected table is present in RPC ReadResponse.
+func verifyReadReceiveMatch(t *testing.T, expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {
 
 	matches := 0
 	for _, table := range received_entry.Entities {
-		if diff := cmp.Diff(expected_update[0].Entity.Entity, table.Entity, protocmp.Transform()); diff != "" {
-			glog.Errorf("Table entry diff (-want +got): \n%s", diff)
-			continue
+		// Added a switch case for the captured table entry.
+		// Switch case to remove the counter_data and meter_config from table entry returned by p4rt server.
+		var tableMap map[string]any
+		switch entry := table.Entity.(type) {
+		case *p4_v1.Entity_TableEntry:
+			ent1, err := json.Marshal(entry)
+			if err != nil {
+				return fmt.Errorf("unable to convert table entry to json: %w", err)
+			}
+			var toMap map[string]any
+			err = json.Unmarshal([]byte(string(ent1)), &toMap)
+			if err != nil {
+				return fmt.Errorf("unable to unmarshal table entry to map: %w", err)
+			}
+			tableMap = toMap["TableEntry"].(map[string]any)
+			delete(tableMap, "meter_config")
+			delete(tableMap, "counter_data")
+		default:
+			t.Logf("Not a table entry: %v", entry)
+		}
+		ent2, err := json.Marshal(expected_update[0].Entity.Entity)
```

Comment:
Can you factor out a helper function, maybe called "tableMapFromEntry" and reuse it in both places? Something like:

```
func tableMapFromEntry(entry *p4_v1.Entity_TableEntry) (map[string]any, error) {
	ent, err := json.Marshal(entry)
	if err != nil {
		return nil, fmt.Errorf("unable to marshal table entry to json: %w", err)
	}
	var toMap map[string]any
	err = json.Unmarshal([]byte(string(ent)), &toMap)
	if err != nil {
		return nil, fmt.Errorf("unable to unmarshal table entry to map: %w", err)
	}
	return toMap["TableEntry"].(map[string]any), nil
}
```

Reply:
Hi Greg, 

I did create a separate helper function but since the received_entry and expected update are not the same type, I have to add a switch case for each. The code now looks like this:

func tableMapFromEntry(entry *p4_v1.Entity_TableEntry) (map[string]any, error) {
	ent, err := json.Marshal(entry)
	if err != nil {
		return nil, fmt.Errorf("unable to marshal table entry to json: %w", err)
	}
	var toMap map[string]any
	err = json.Unmarshal([]byte(string(ent)), &toMap)
	if err != nil {
		return nil, fmt.Errorf("unable to unmarshal table entry to map: %w", err)
	}
	return toMap["TableEntry"].(map[string]any), nil
}

// Function to compare and check if the expected table is present in RPC ReadResponse.
func verifyReadReceiveMatch(t *testing.T, expected_update []*p4_v1.Update, received_entry *p4_v1.ReadResponse) error {

	matches := 0
	for _, table := range received_entry.Entities {
		// Added a switch case for the captured table entry.
		// Switch case to remove the counter_data and meter_config from table entry returned by p4rt server.
		var tableMap map[string]any
		switch entry := table.Entity.(type) {
		case *p4_v1.Entity_TableEntry:
			tableMap, err := tableMapFromEntry(entry)
			if err != nil {
				return fmt.Errorf("unable to convert table entry to json: %w", err)
			}
			delete(tableMap, "meter_config")
			delete(tableMap, "counter_data")
		default:
			t.Logf("Not a table entry: %v", entry)
		}
		
		var tableMap1 map[string]any
		switch expected_entry := expected_update[0].Entity.Entity.(type) {
		case *p4_v1.Entity_TableEntry:
			tableMap1, err := tableMapFromEntry(expected_entry) 
			if err != nil {
				return fmt.Errorf("Unable to convert table entry to json: %w", err)
			}
			fmt.Println(tableMap1)
		default:
			t.Logf("Not a table entry: %v", expected_entry)
		}
		

		if cmp.Equal(tableMap, tableMap1) {
			t.Logf("Table match succesful")
			matches += 1
		}

	}
	if matches == 0 {
		return errors.New("No matches found")
	}
	return nil
}

Please let me know if this looks okay. Thanks.

Reply:
I think you should factor out even more into the helper. If you change the parameters of tableMapFromEntry to (*testing.T, *p4_v1.Entity), it should collapse to something closer to:

```
tableMap, err  := tableMapFromEntry(t, table)
if err != nil {
    return err
}
delete(tableMap, "meter_config")
delete(tableMap, "counter_data")
tableMap1, err  := tableMapFromEntry(t, expected_update[0].Entity)
if err != nil {
    return err
}
```

Reply:
Hi Greg, 

I will make those changes in the next commit.